test/expected/timestamp.out

```diff
@@ -1243,6 +1243,9 @@ ERROR:  interval can't combine months with minutes or hours
 -- timestamp is less than the default 'origin' value
 SELECT timescaledb_experimental.time_bucket_ng('1 day', '1999-01-01 12:34:56 MSK' :: timestamptz, timezone => 'MSK');
 ERROR:  origin must be before the given date
+-- 'origin' is not a first day of the month after converting from provided time zone to UTC
+select timescaledb_experimental.time_bucket_ng('1 month', '2021-07-12 12:34:56 Europe/Moscow' :: timestamptz, origin => '2021-06-01 00:00:00 Europe/Moscow' :: timestamptz, timezone => 'UTC');
+ERROR:  origin must be the first day of the month
```

Comment:
It would be helpful to add a hint explaining why it fails.
Something along the lines of:
errhint('origin must be first day of the month after applying timezone conversions specified by `timezone` argument).

The error messages usually specify the argument value that causes the failure.
e.g. :  elog(ERROR, "invalid time_bucket type \"%s\"", format_type_be(timestamp_type));

So for this error, I would expect something like:
`origin '2021-06-01 00:00:00 Europe/Moscow' must be the first day of the month.`
along with a hint why it is not so.

We have a similar issue with the error message: `origin must be before given date`. It doesn't give additional info on what origin or date is being referred to.

Reply:
Thanks for the feedback, Gayathri! I did what I could without refactoring the rest of the code (and the tests) too much. I suggest later we invest some time into reviewing and improving all the error messages of time_bucket_ng(). There was not too much effort put into them at this point. I agree that this case was especially confusing.